communities.go

```diff
@@ -0,0 +1,197 @@
+			return ctx.Send([]byte("Error listing communities"), http.StatusInternalServerError)
+		}
+	default:
+		// otherwise, list all communities
+		if dbCommunities, err = v.db.ListCommunities(); err != nil {
+			return ctx.Send([]byte("Error listing communities"), http.StatusInternalServerError)
+		}
+	}
+	if len(dbCommunities) == 0 {
+		return ctx.Send([]byte("No communities found"), http.StatusNotFound)
+	}
+	communities := CommunityList{
+		Communities: []*Community{},
+	}
+	for _, c := range dbCommunities {
+		// get admin profiles from the database
+		admins := []*User{}
+		for _, admin := range c.Admins {
+			user, err := v.db.User(admin)
+			if err != nil {
+				if err == farcasterapi.ErrNoDataFound {
+					continue
+				}
+				return ctx.Send([]byte(err.Error()), apirest.HTTPstatusInternalErr)
+			}
+			admins = append(admins, &User{
+				FID:         user.UserID,
+				Username:    user.Username,
+				DisplayName: user.Displayname,
+				// Avatar:      user.Avatar,
+			})
+		}
+		// get census channel or addresses based on the type
+		cAddresses, cChannel, err := v.censusChannelOrAddresses(ctx.Request.Context(), c.Census)
+		if err != nil {
+			if err == farcasterapi.ErrChannelNotFound {
+				return ctx.Send([]byte("Census channel not found"), http.StatusNotFound)
+			}
+			return ctx.Send([]byte(err.Error()), apirest.HTTPstatusInternalErr)
+		}
+		// add community to the list
+		communities.Communities = append(communities.Communities, &Community{
+			ID:              c.ID,
+			Name:            c.Name,
+			LogoURL:         c.ImageURL,
+			Admins:          admins,
+			Notifications:   c.Notifications,
+			CensusName:      c.Census.Name,
+			CensusType:      c.Census.Type,
+			CensusAddresses: cAddresses,
+			CensusChannel:   cChannel,
+			Channels:        c.Channels,
+		})
+	}
+	res, err := json.Marshal(communities)
+	if err != nil {
+		return ctx.Send([]byte("Error encoding communities"), http.StatusInternalServerError)
+	}
+	return ctx.Send(res, http.StatusOK)
+}
+
+func (v *vocdoniHandler) communityHandler(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	// get community id from the URL and parse to int
+	communityID := ctx.URLParam("communityID")
+	if communityID == "" {
+		return ctx.Send([]byte("No community ID provided"), http.StatusBadRequest)
+	}
+	id, err := strconv.Atoi(communityID)
+	if err != nil {
+		return ctx.Send([]byte("Invalid community ID"), http.StatusBadRequest)
+	}
+	// get the community from the database by its id
+	dbCommunity, err := v.db.Community(uint64(id))
+	if err != nil {
+		return ctx.Send([]byte("Error getting community"), http.StatusInternalServerError)
+	}
+	if dbCommunity == nil {
+		return ctx.Send([]byte("Community not found"), http.StatusNotFound)
+	}
+	// get admin profiles for the community
+	admins := []*User{}
+	for _, admin := range dbCommunity.Admins {
+		user, err := v.db.User(admin)
+		if err != nil {
+			if err == farcasterapi.ErrNoDataFound {
+				continue
```

Comment:
Do not silent the errors please, it's ok to keep going but print it at least as a Warning on the logs.